libr/anal/var.c

```diff
@@ -239,10 +240,23 @@ R_API RAnalVar *r_anal_var_get_byname (RAnal *anal, RAnalFunction *fcn, char kin
 			break;
 		}
 	}
+	
 	if (!var || strcmp (name, var->name)) {
+		r_list_free (var_list);
 		return 0;
 	}
-	return  var;
+
+	av = R_NEW0 (RAnalVar);
```

Comment:
why copying copy of var instead of returning it.
I guess that is what caused the bug